contrib/urls-maintenance.sh

```diff
@@ -0,0 +1,113 @@
+#!/bin/sh
+# operates various maintenance tasks on urls and cache.db for newsboat:
+# - convert http to https in urls file when https feed is available
+# - convert feeds urls to their new forwarded urls for 3xx http codes
+# - feeds urls not returning a valid http code 200 are tagged "_dead"
+# cache.db is also updated to refer to the new url feeds so all read/unread articles and flags are preserved
+
+# TODO
+# implement additional checks for "valid" feeds returning 200 http code: 
+# https://www.linuxjournal.com/content/parsing-rss-news-feed-bash-script
+#	is it returning valid rss?
+#	when was the feed last updated?
+#	sort valid feeds by last updated
+#	tag feed "abandoned" when feed not updated for more than 1 year
+
+
+#newsboat urls file and cache locations
+u="$HOME/.config/newsboat/urls"
+db="$HOME/.config/newsboat/cache.db"
+#curl timeout for testing URLs
+timeout=10
+tagdead="_dead"
+
+# shuf (GNU coreutils) randomises the urls list, this avoids querying the same domains too fast, assuming urls are grouped by domains or alphasorted in the urls file
+requirements=( newsboat curl sqlite3 sed grep awk head shuf command )
+for app in ${requirements[*]}
+do
+	command -v "$app" >/dev/null 2>&1 || { echo >&2 "$app is required but it's not installed or it's not in your PATH. Aborting."; exit 1; }
+done
+
+if [ ! -f "$u" ]; then
+	echo "$u not found. edit the path/filename for urls file"; exit
+fi
+if [ ! -f "$db" ]; then
+	echo "$db not found. edit the path/filename for cache.db"; exit
+fi
+if [ -f "$db.lock" ]; then
+	echo "newsboat is still running. Close it first then try again"; exit
+fi
+
+cp "$db" "$db.bak-$(date +%FT%T)"
+cp "$u" "$u.bak-$(date +%FT%T)"
+
+feeds=$(wc -l "$u" | awk '{print $1}')
+i=0
+# check all feeds return valid http codes
+for url in $(shuf "$u" | awk '{print $1}')
+do
+	i=$((i+1))
```

Comment:
[nitpick] Given we're going full-on bash let's make it prettier. 
```suggestion
	let i=i+1
```

Reply:
the script is going posix compliant. at least i'm trying.